hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -7322,13 +7323,14 @@ public static HRegion openHRegion(final Configuration conf, final FileSystem fs,
    * @param rsServices An interface we can request flushes against.
    * @param reporter An interface we can report progress against.
    * @return new HRegion
+   * @throws NullPointerException if {@code info} is {@code null}
    */
   public static HRegion openHRegion(final Configuration conf, final FileSystem fs,
       final Path rootDir, final Path tableDir, final RegionInfo info, final TableDescriptor htd,
       final WAL wal, final RegionServerServices rsServices,
       final CancelableProgressable reporter)
       throws IOException {
-    if (info == null) throw new NullPointerException("Passed region info is null");
+    Objects.requireNonNull(info, "RegionInfo cannot be null");
```

Comment:
Should we add @NotNull annotations given you are in here? Could be a follow-on no problem.

Reply:
As long as there's a helpful JavaDoc, that should be good enough for now.